### DISCUSSION TITLE
Standalone Pi Image

### DISCUSSION THREAD

**Discussion by VladimirKuletski** [Berny23/LD-ToyPad-Emulator#188] [Ideas]
Here is what I've got: [image link](https://disk.yandex.ru/d/j2R8p8SJxCc-jQ)
Prebuilt image for Raspberry Pi Zero W. It includes all components to work w/o internet connection.
Version includes ability to render toytag images. You'll have to create/get them and upload yourself.

Use Raspberry Pi Imager to burn image to SD card.
Choose "Raspberry Pi Zero " in board settings
![image](https://github.com/Berny23/LD-ToyPad-Emulator/assets/31250777/61ea904a-9148-4398-a56e-3e6e737d020c)
Under "Operating system" choose "Use custom" and select downloaded zip file.
Under "Storage" choose your SD card

Should look like this:
![Settings view](https://github.com/Berny23/LD-ToyPad-Emulator/assets/31250777/23bf66ac-6182-429b-a83c-4b5106381577)
Click "Next" - "No" for customization
![Customization](https://github.com/Berny23/LD-ToyPad-Emulator/assets/31250777/399251a4-f06e-4473-bd50-204b0328f034)
"Yes" at "Warning"
![Confirmation](https://github.com/Berny23/LD-ToyPad-Emulator/assets/31250777/5030206f-715a-4985-a460-51179d0e478a)
Remove SD card from card reader after it's done writing.

Connect Pi to console, turn on console and wait for 2-3 minutes.
After initial boot Pi will create open access point `LD`. Use your phone/tablet/PC to connect and you'll be redirected to portal. Please keep in mind that some phones might drop WiFi connection because of `No Internet connectivity`
Use direct link `http://192.168.4.1:3000` if redirect failed.

Image/Pi should not require any additional configuration step.

**Comment by szechman** [comment]
I have tried your image and have a lot of activity (green light on pi flashes alot) but I never get a ssid called LD to show up in my phone or my computer. Any help would be appreciated.

**Reply by VladimirKuletski** [reply]
Just to doublecheck: you are using Pi Zero W, not Pi Zero 2 W. Correct?

Will also burn image to card and check it one more time today.

**Comment by szechman** [comment]
![20240808_074806_HDR](https://github.com/user-attachments/assets/6766b417-19f4-4341-861a-935eb4b81570)

Think I read this correctly. Thank you for helping.

**Reply by VladimirKuletski** [reply]
I hope: ` "No" for customization` - this item was also taken into account

Followed the guide above. Re-downloaded image, flashed to new card.
After 1m44s my phone shows `LD` SSID
![Parameters](https://github.com/user-attachments/assets/f6e35ab9-b664-481c-afb7-d26129654bdb)
![Flashed image](https://github.com/user-attachments/assets/0a3bbfbb-ec06-4647-b82b-b5a762e1ba1c)
![SSID3](https://github.com/user-attachments/assets/e9f9403e-0dda-46a6-80d9-6024c363f532)

Sorry, no ideas other than trying different card and giving it one more shot.

**Comment by szechman** [comment]
Looking at your 2nd image it looks like you flashed the zip file. I
unzipped it first. Was i not supposed to do that?

**Reply by VladimirKuletski** [reply]
You can do it either way.

RPi Imager should be smart enough to handle this by itself.

**Comment by VladimirKuletski** [comment]
Can anybody else confirm if it worked? Or it "works on my laptop" only?

**Comment by szechman** [comment]
I tried deleting all the partitions on my sd card and reflashing with the
zip file. I then used a different pi (same version as I had 2). Made sure
it was a good data cable plugged into the second (middle) port on the pie
and still the same. Lots of flashing led on the pie but no ssid. I even
checked a different cell phone. Is there anything I can check on the SD
card?

**Comment by szechman** [comment]
I got it working. Not too sure what I did. I reflashed again, made sure i was connected to the ps3 with the middle usb and a power brick to the top usb and after about 2 minutes i had the LD ssid on my phone. I did have to "confirm" that I wanted to connect because of "no internet with this connection" but after that it works evey time. Thank you for your help and a much simplified way of doing this.

**Reply by VladimirKuletski** [reply]
Awesome news! Haven't heard any confirmation that it's actually working - now we have it!
Hope it will help you with the game!

P.S. Will ask for Pi 4 version. Hope it will work the same easy way!

**Comment by Galactic-Hot-Dog** [comment]
Hey, sorry if it might sound dumb, as I'm fairly new to this whole subject but how would I go on a about safely shutting down the pi when not used?

**Reply by Berny23** [reply]
It should work with this command: `sudo poweroff`

This is the same as clicking Shutdown in a graphical session.

**Reply by Galactic-Hot-Dog** [reply]
So do I always have to connect via ssh on pc before turning off the toy pad emulator, or can I just let it turn off with my PS5? 

Thanks already!

**Reply by Berny23** [reply]
Your console does not send any shutdown command because it sees the computer as a toy pad, not Linux. It just turns off the power supply for the USB ports.

Yes, to properly shut down, you have to connect via ssh. Compared to Windows, Linux is pretty robust when it comes to unexpected shutdowns, so you'll be probably safe not doing this, as there are no large data transfers to your SD card going on that could be lost.

**Reply by Galactic-Hot-Dog** [reply]
Thank you for your indepth answer and for taking the time!

I'm going to use your guide and setup the emulator in the following week when all things arrive.

And thanks for this project, because I'm able to play the game again without the need to buy any figures nor replicate nfc tags.

**Reply by VladimirKuletski** [reply]
I don't do proper shutdowns and the card hasn't been corrupted yet. Worst case scenario - I'll have to write an image one more time. 
You can put a button connected to one of the GPIO and setup the script to monitor it.
People here are struggling to write image so adding button stuff will make it even more complex for beginners.

**Reply by Galactic-Hot-Dog** [reply]
Okay, so that would need some soldering I assume or just plug it in somehow?

**Reply by VladimirKuletski** [reply]
There are plenty `pi buttons` on Amazon and Ali with crimped connectors. You'll need to just connect them. But you might also need a case to make it look better.

**Reply by cjunisl** [reply]
Would it be possible to add a shutdown button on the webpage?"

**Comment by RamsesUTL** [comment]
Works like a charme (Pi Zero W & WiiU). Thanks for the image!

**Comment by miraliveonce** [comment]
raspberry pi zero 2w is working, thank you!!!

**Reply by Tatshino** [reply]
How'd you get your pi zero 2 w working? Struggling a bit with mine and it doesn't seem to start the ssid for me to find

**Comment by marcp87** [comment]
i might be a little late but i can connect to the pi via network okay but when i go to start the game it tells me that i still need to plug the toypad in? just in the case it’s needed im playing on ps5 with the pi plugged in the back, using the usb port on the pi.

**Reply by VladimirKuletski** [reply]
- Please double check that correct port is used. 
https://github.com/Berny23/LD-ToyPad-Emulator/issues/45#issuecomment-1022670064
- Please check that cable supports data transfer by using it with different device

**Comment by PurpleGuy635** [comment]
hi does this image also work for raspberry pi zero 2W?

**Reply by VladimirKuletski** [reply]
Sorry. Never checked and don't have one in my collection.
Looking for someone who can create image for Pi 4 - code definitely works there

**Comment by cjunisl** [comment]
For some reason, my Pi doesn’t connect to my router, and I couldn’t use it in gadget mode directly on my PC. I was thinking of various ways to make everything work, and by chance, I discovered this perfect solution. Thank you!!! now I can finally play. 🙏
